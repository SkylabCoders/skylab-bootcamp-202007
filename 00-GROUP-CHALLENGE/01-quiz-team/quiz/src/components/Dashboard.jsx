import React, { useState, useEffect } from 'react';
import Item from './Item';
import './../css/dashboard.scss';
import gameStore from './../stores/gameStore';
import {loadThemes, loadTopThemes, loadSessionSet} from './../actions/gameActions';


function Dashboard() {
    const [themesList, setThemesList] = useState([]);
    const [topThemesList, setTopThemesList] = useState([]);
    const [sessionSet, setSessionSet] = useState([]);

    useEffect(() => {
        gameStore.addChangeListener(onChangeThemes);
        if (themesList.length === 0) { loadThemes() }
        return () => { gameStore.removeChangeListener(onChangeThemes); }
    }, []);

    useEffect(() => {
        gameStore.addChangeListener(onChangeTopThemes);
        if (topThemesList.length === 0) { loadTopThemes() }
        return () => { gameStore.removeChangeListener(onChangeTopThemes); }
    }, []);

    useEffect(()=>{
        gameStore.addChangeListener(onChangeSessionSet);
        if(sessionSet.length === 0){loadSessionSet()}
        return ()=>{gameStore.removeChangeListener(onChangeSessionSet);}
    }, [sessionSet]);
    
    function onChangeThemes(){
        setThemesList(gameStore.getThemes());
    }

    function onChangeTopThemes() {
        setTopThemesList(gameStore.getTopThemes());
    }

    function onChangeSessionSet(){
        setSessionSet(gameStore.getSessionSet());
    }

    return (
        <>
            <div className="dashboard__container">
                <div className="dashboard__themes">
                    <h2 className="section__title">THEMES</h2>
                    
                        <ul className="themes__grid">
                            {themesList.map((theme) => (
                                <li key={theme.title}>
                                        <Item themeTitle={theme.title} themeImgurl={theme.url} themeId={theme.id} themeSlug={theme.slug}/>
                                </li>
                            ))}
                        </ul>
                </div>
                <div className="dashboard__topThemes">
                    <h2 className="section__title">TOP THEMES</h2>
                    <ul className="topThemes__grid">
                        {topThemesList.map((topTheme) => (
                            <li key={topTheme.title}>
                                <Item themeTitle={topTheme.title} themeImgurl={topTheme.url} themeId={topTheme.id} themeSlug={topTheme.slug}/>
                            </li>
                        ))}
                    </ul>
                </div>
            </div>
        </>
    )
}

export default Dashboard;