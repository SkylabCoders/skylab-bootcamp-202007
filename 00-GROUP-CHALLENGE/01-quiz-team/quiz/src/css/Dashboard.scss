.dashboard__container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.section__title {
  color: white;
  text-align: center;
}

.topThemes__grid,
.themes__grid {
  list-style-type: none;
  display: grid;
  justify-items: center;
  grid-template-columns: 1;
  padding: 0;
}

.dashboard__topThemes {
  display: none;
}

@media (min-width: 680px) and (max-width: 1024px) {
  .topThemes__grid,
.themes__grid {
    grid-template-columns: repeat(3, 1fr);
  }

  .dashboard__topThemes {
    display: contents;
  }
}
@media (min-width: 1024px) {
  .topThemes__grid,
.themes__grid {
    grid-template-columns: repeat(5, 1fr);
  }

  .dashboard__topThemes {
    display: contents;
  }
}

/*# sourceMappingURL=Dashboard.scss.map */
