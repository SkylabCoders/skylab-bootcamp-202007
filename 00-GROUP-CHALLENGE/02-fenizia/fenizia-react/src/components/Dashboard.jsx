import React from "react";
import Slider from "./Slider";
import Cover from './Cover'


function Dashboard() {
  return (
    <>
        <Cover />
        <Slider order="novedades" url="/novedades" text="NOVEDADES" />
        <Slider order="superventas" url="/superventas" text="SUPERVENTAS" />
    </>
  );
}

export default Dashboard;
