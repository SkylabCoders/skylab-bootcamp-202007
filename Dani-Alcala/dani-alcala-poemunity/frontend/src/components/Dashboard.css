.dashboard {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  margin-top: 10px;
}

.dashboard__accordion {
  display: none;
}

.dashboard__ranking {
  display: none;
}

@media screen and (min-width: 1200px) {
  .dashboard {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
        -ms-flex-direction: row;
            flex-direction: row;
  }
  .dashboard__accordion {
    display: block;
    width: 15%;
    margin-left: 3%;
    margin-right: 3%;
  }
  .dashboard__list {
    width: 65%;
    border-left: 2px solid #eee;
  }
  .dashboard__ranking {
    display: block;
    width: 15%;
    margin-right: 4%;
  }
}
/*# sourceMappingURL=Dashboard.css.map */