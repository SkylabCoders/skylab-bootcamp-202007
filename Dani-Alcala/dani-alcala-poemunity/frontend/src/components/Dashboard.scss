.dashboard {
    display: flex;
    flex-direction: column;
    margin-top: 10px;
}

.dashboard__accordion {
    display: none;
}

.dashboard__ranking {
    display: none;
}

@media screen and (min-width: 1200px) {
    .dashboard {
        display: flex;
        flex-direction: row;
    }
    
    .dashboard__accordion {
        display: block;
        width: 15%;
        margin-left: 3%;
        margin-right: 3%;
    }

    .dashboard__list {
        width: 65%;
        border-left: 2px solid #eee;
    }

    .dashboard__ranking {
        display: block;
        width: 15%;
        margin-right: 4%;
    }
    
}