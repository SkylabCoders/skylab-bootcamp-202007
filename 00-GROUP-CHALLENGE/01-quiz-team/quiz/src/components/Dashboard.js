import React, { useState, useEffect } from 'react';
import Welcome from './Welcome';
import Item from './Item';
import './../css/Dashboard.css';
import gameStore from './../stores/gameStore';
import {loadThemes, loadTopThemes} from './../actions/gameActions';

function Dashboard(){
    const [themesList, setThemesList] = useState([]);
    const [topThemesList, setTopThemesList] = useState([]);

    useEffect(()=>{
        gameStore.addChangeListener(onChangeThemes);
        if(themesList.length === 0){loadThemes()};
        return ()=>{gameStore.removeChangeListener(onChangeThemes);}
    }, []);

    useEffect(()=>{
        gameStore.addChangeListener(onChangeTopThemes);
        if(topThemesList.length === 0){loadTopThemes()};
        return ()=>{gameStore.removeChangeListener(onChangeTopThemes);}
    }, []);
    
    function onChangeThemes(){
        setThemesList(gameStore.getThemes());
    }

    function onChangeTopThemes(){
        setTopThemesList(gameStore.getTopThemes());
    }

    return (
        <>
            <div className="dashboard__container">
                <div className="dashboard__welcome">
                    <Welcome />
                </div>
                <div className="dashboard__themes">
                    <h2>THEMES</h2>
                    <ul className="themes__grid">
                    {themesList.map((theme)=>(
                        <li key={theme.title}>
                            <Item themeTitle={theme.title} themeImgurl={theme.url} themeId={theme.id}/>
                        </li>
                    ))}
                    </ul>
                </div>
                <div className="dashboard__topThemes">
                    <h2>TOP THEMES</h2>
                    <ul className="topThemes__grid">
                    {topThemesList.map((topTheme)=>(
                        <li key={topTheme.title}>
                            <Item themeTitle={topTheme.title} themeImgurl={topTheme.url} themeId={topTheme.id}/>
                        </li>
                    ))}
                    </ul>
                </div>
            </div>
        </>
    )
}

export default Dashboard;