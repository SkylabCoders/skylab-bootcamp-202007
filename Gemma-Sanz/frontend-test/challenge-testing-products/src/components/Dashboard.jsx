import React from 'react';
import ProductList from './ProductList';
import CartList from './CartList';
import './productList.css';

function Dashboard() {
	return (
		<div className="main__container">
			<ProductList />
			<CartList />
		</div>
	);
}

export default Dashboard;
