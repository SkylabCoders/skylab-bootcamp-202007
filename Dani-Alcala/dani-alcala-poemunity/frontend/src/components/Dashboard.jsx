import React from "react";
import "./List.scss";
import "./Detail.scss";
import "../App.scss";
import "./Dashboard.scss";
import Accordion from "./SimpleAccordion";
import Ranking from "./Ranking";
import List from "./List";


function Dashboard(props) {
  return (
    <main className="dashboard">
      <div className="dashboard__accordion">
      <Accordion />
      </div>
      <div className="dashboard__list">
      <List {...props}/>
      </div>
      <div className="dashboard__ranking">
      <Ranking />
      </div>
    </main>
  );
}
export default Dashboard;
