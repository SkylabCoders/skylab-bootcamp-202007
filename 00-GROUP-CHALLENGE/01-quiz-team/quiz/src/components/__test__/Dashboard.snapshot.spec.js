import renderer from 'react-test-renderer'
import Dashboard from '../../components/Dashboard'
import React from 'react'

describe('Dashboard snapshot', ()=>{
    const treeDashboard = renderer.create(
        <Dashboard />
    );

    it('should match', ()=>{
        expect(treeDashboard).toMatchSnapshot();
    })
})
