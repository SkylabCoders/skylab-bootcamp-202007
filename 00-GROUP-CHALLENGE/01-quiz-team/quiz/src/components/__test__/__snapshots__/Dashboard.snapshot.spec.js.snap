// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard snapshot should match 1`] = `
<div
  className="dashboard__container"
>
  <div
    className="dashboard__themes"
  >
    <h2
      className="section__title"
    >
      THEMES
    </h2>
    <ul
      className="themes__grid"
    />
  </div>
  <div
    className="dashboard__topThemes"
  >
    <h2
      className="section__title"
    >
      TOP THEMES
    </h2>
    <ul
      className="topThemes__grid"
    />
  </div>
</div>
`;
